File: examples/python-bot/bot.py
import asyncio
import os

from aiogram import Bot, Dispatcher
from aiogram.filters import Command
from aiogram.types import Message


TOKEN = os.environ["TELEGRAM_BOT_TOKEN"]
dp = Dispatcher()


@dp.message(Command("start"))
async def start(message: Message) -> None:
    await message.answer("Hello from an AdBotHost Python bot.")


@dp.message()
async def echo(message: Message) -> None:
    await message.answer(f"Echo: {message.text}")


async def main() -> None:
    bot = Bot(TOKEN)
    await dp.start_polling(bot)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/node-bot/bot.js
import { Telegraf } from "telegraf";

const token = process.env.TELEGRAM_BOT_TOKEN;

if (!token) {
  throw new Error("TELEGRAM_BOT_TOKEN is required");
}

const bot = new Telegraf(token);

bot.start((ctx) => ctx.reply("Hello from an AdBotHost Node.js bot."));
bot.on("text", (ctx) => ctx.reply(`Echo: ${ctx.message.text}`));

bot.launch();

process.once("SIGINT", () => bot.stop("SIGINT"));
process.once("SIGTERM", () => bot.stop("SIGTERM"));
